include/nuttx/fs/hostfs.h

```diff
@@ -106,8 +106,13 @@ typedef int16_t      nuttx_uid_t;
 typedef uint16_t     nuttx_dev_t;
 typedef uint16_t     nuttx_ino_t;
 typedef uint16_t     nuttx_nlink_t;
+#ifdef CONFIG_FS_LARGEFILE
+typedef int64_t      nuttx_off_t;
```

Comment:
Requires CONFIG_LONG_LONG.  This will break all ZiLOG platforms since the ZiLOG compiler does not support int64_t types.

Reply:
Yes, I found this issue before, but it's hard to let FS_LARGEFILE depends on LONG_LONG since LONG_LONG isn't a Kconfig option, but auto define in include/nuttx/compiler.h:
https://github.com/apache/incubator-nuttx/blob/master/include/nuttx/compiler.h#L169
Do you have some better method to resolve this issue?

Reply:
Ok, we can check both CONFIG_FS_LARGEFILE and CONFIG_LONG_LONG to fix this issue. Please review the new version.

Reply:
A cleaner solution would be to move the selection of CONFIG_HAVE_LONG_LONG to a Kconfig file.  In fact, all of compiler.h could be move to Kconfig files.  This solution is fine with me, however.